client/src/components/SchoolQuerySearch.js

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useState } from "react";
+
+export const SchoolQuerySearch = ({ value, handleSearch }) => {
+  const [query, setQuery] = useState("");
+  const [results, setResults] = useState([]);
+  const hasResults = results.length > 0;
+
+  const [timerId, setTimerId] = useState(null);
+  const TIMER_DELAY = 500; // milliseconds
+
+  useEffect(() => {
+    // Cleanup the timer on component unmount
+    return () => {
+      if (timerId) {
+        clearTimeout(timerId);
+      }
+    };
+  }, [timerId]);
+
+  // initialize school if one exists
+  useEffect(() => {
+    // reset query before checking if theres a previous value
+    setQuery("");
+
+    // check if there was a previous value
+    // if there was, set query to it
+    if (value) {
+      setQuery(value);
+    }
+  }, [value]);
+
+  const fetchSchools = async (query) => {
+    // edge case: empty query
+    if (query === "") return;
+
+    try {
+      const response = await fetch(
+        `http://universities.hipolabs.com/search?name=${query}`,
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        },
+      );
+
+      if (!response.ok) {
+        // Check if the response has JSON content
+        if (
+          response.headers.get("content-type")?.includes("application/json")
+        ) {
+          const errorData = await response.json();
+          throw new Error(`${errorData.error}`);
+        } else {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+      }
+
+      const data = await response.json();
+
+      // Remove duplicates based on the "name" property
+      const uniqueResults = data.reduce((unique, school) => {
+        if (!unique.find((item) => item.name === school.name)) {
+          unique.push(school);
+        }
+        return unique;
+      }, []);
+
+      console.log("data:", data);
+      console.log("query:", query);
```

Comment:
Should remove these console logs before pushing to prod

Reply:
^